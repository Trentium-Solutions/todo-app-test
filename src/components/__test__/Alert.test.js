import Alert from '../Alert'
import {createRoot} from 'react-dom/client'
import TestRenderer from 'react-test-renderer'
import {render, screen} from '@testing-library/react'

const alert = {
  type: 'success',
  message: 'Displayed Successfully',
}
describe('alert test suite', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    const root = createRoot(div)
    root.render(<Alert alert={alert}></Alert>)
  })

  it('renders alert correctly', () => {
    render(<Alert alert={alert}></Alert>)
    expect(screen.getByTestId('alert')).toBeTruthy()
  })

  it('matches snapshot alert', () => {
    const tree = TestRenderer.create(<Alert alert={alert}></Alert>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
