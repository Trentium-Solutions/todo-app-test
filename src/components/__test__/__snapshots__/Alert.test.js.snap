// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert test suite matches snapshot alert 1`] = `
<div
  style={
    Object {
      "height": "50px",
    }
  }
>
  <div>
    <div
      className="alert alert-success alert-dismissible fade show"
      data-testid="alert"
      role="alert"
    >
      <strong>
        Success
        : 
      </strong>
       
      Displayed Successfully
    </div>
  </div>
</div>
`;
